include/customizer/cell_customizer.hpp

```diff
@@ -0,0 +1,148 @@
+#ifndef OSRM_CELLS_CUSTOMIZER_HPP
+#define OSRM_CELLS_CUSTOMIZER_HPP
+
+#include "util/binary_heap.hpp"
+#include "util/cell_storage.hpp"
+#include "util/multi_level_partition.hpp"
+
+#include <boost/thread/tss.hpp>
+#include <unordered_set>
+
+namespace osrm
+{
+namespace customizer
+{
+
+class CellCustomizer
+{
+
+  public:
+    CellCustomizer(const osrm::util::MultiLevelPartition &partition) : partition(partition) {}
+
+    template <typename GraphT>
+    void Customize(const GraphT &graph,
+                   osrm::util::CellStorage &cells,
+                   osrm::util::LevelID level,
+                   osrm::util::CellID id)
+    {
+        auto cell = cells.GetCell(level, id);
+        auto destinations = cell.GetDestinationNodes();
+
+        // for each source do forward search
+        for (auto source : cell.GetSourceNodes())
+        {
+            std::unordered_set<NodeID> destinations_set(destinations.begin(), destinations.end());
```

Comment:
Here it is just a data structure with fast erase operation to stop search if all destination nodes are reached.